core/src/dataio.cxx

```diff
@@ -24,7 +24,11 @@ g3_istream_from_path(boost::iostreams::filtering_istream &stream,
 	if (boost::algorithm::ends_with(path, ".gz"))
 		stream.push(boost::iostreams::gzip_decompressor());
 	if (boost::algorithm::ends_with(path, ".bz2"))
+#ifdef Boost_BZIP2_FOUND
```

Comment:
I think you need to add this directive to the target compile definitions list for the core library, otherwise the source code doesn't know about it.

Reply:
Can you suggest a fix? I don't really know how this works.

Reply:
```
diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index 8efa3cb..4c87b51 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -35,6 +35,10 @@ if(FLAC_FOUND)
        target_link_libraries(core ${FLAC_LIBRARIES})
 endif()

+if(Boost_BZIP2_FOUND)
+        target_compile_definitions(core PRIVATE -DBoost_BZIP2_FOUND)
+endif()
+
 link_python_dir()

 add_spt3g_program(bin/spt3g-dump)
```